cpp/cmake_modules/FindgRPCAlt.cmake

```
@@ -36,7 +36,7 @@ if(GRPCPP_PC_FOUND)
   # gRPC's pkg-config file neglects to specify pthreads.
   find_package(Threads REQUIRED)
   if(ARROW_GRPC_USE_SHARED)
-    set(GRPCPP_LINK_LIBRARIES ${GRPCPP_PC_LINK_LIBRARIES} Threads::Threads)
+    list(APPEND GRPCPP_LINK_LIBRARIES ${GRPCPP_PC_LINK_LIBRARIES})
```

Could you use `set()` instead of `list(APPEND)` because we don't use any pre-defined `GRPCPP_LINK_LIBRARIES` here?

```suggestion
    set(GRPCPP_LINK_LIBRARIES ${GRPCPP_PC_LINK_LIBRARIES})
```

---

It will create the list if it does not exists but I see how using set is more expressive (also less chars :D).